### PR TITLE
rewrite statistics tests not to use numpy_cupy_raises()

### PR DESCRIPTION
**Pull request by rushabh-v**
ready for #3098 

Found some `numpy_cupy_raises` calls in `test_histogram` while investigating for my gsoc project.
This PR rewrites the tests with `numpy_cupy_raises` calls in the `cupy/statistics/` directory.

### CODE REVIEW

File: tests/cupy_tests/statics_tests/test_histogram.py [hunk 1]
```diff
@@ -294,10 +301,11 @@ def test_bincount_zero(self, xp, dtype):
...
     @for_all_dtypes_bincount()
-    @testing.numpy_cupy_raises()
-    def test_bincount_too_small_minlength(self, xp, dtype):
-        x = testing.shaped_arange((3,), xp, dtype)
-        return xp.bincount(x, minlength=-1)
+    def test_bincount_too_small_minlength(self, dtype):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((3,), xp, dtype)
+            with pytest.raises(ValueError):
+                xp.bincount(x, minlength=-1)
```
Comment (rushabh-v) [inline tests/cupy_tests/statics_tests/test_histogram.py:308]: This test fails with the below log.
```python
_______________ TestHistogram.test_bincount_too_small_minlength ________________

self = <cupy_tests.statics_tests.test_histogram.TestHistogram testMethod=test_bincount_too_small_minlength>
dtype = <class 'numpy.float16'>

    @for_all_dtypes_bincount()
    def test_bincount_too_small_minlength(self, dtype):
        for xp in (numpy, cupy):
            x = testing.shaped_arange((3,), xp, dtype)
            with pytest.raises(ValueError):
>               xp.bincount(x, minlength=-1)

tests/cupy_tests/statics_tests/test_histogram.py:308: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (array([1., 2., 3.], dtype=float16),), kwargs = {'minlength': -1}
relevant_args = (array([1., 2., 3.], dtype=float16), None)

>   ???
E   TypeError: Cannot cast array data from dtype('float16') to dtype('int64') according to the rule 'safe'

<__array_function__ internals>:6: TypeError
----------------------------- Captured stdout call -----------------------------
dtype is <class 'numpy.float16'>
```

File: tests/cupy_tests/statics_tests/test_histogram.py [hunk 2]
```diff
@@ -263,29 +266,33 @@ def test_bincount_with_minlength(self, xp, dtype):
...
-    def test_bincount_invalid_weight_length(self, xp, x_type, w_type):
-        x = testing.shaped_arange((1,), xp, x_type)
-        w = testing.shaped_arange((2,), xp, w_type)
-        return xp.bincount(x, weights=w)
+    def test_bincount_invalid_weight_length(self, x_type, w_type):
+        for xp in (numpy, cupy):
+            x = testing.shaped_arange((1,), xp, x_type)
+            w = testing.shaped_arange((2,), xp, w_type)
+            with pytest.raises(ValueError):
+                xp.bincount(x, weights=w)
```
Comment (rushabh-v) [inline tests/cupy_tests/statics_tests/test_histogram.py:274]: this test fails with the below log.
```python
______________ TestHistogram.test_bincount_invalid_weight_length _______________

self = <cupy_tests.statics_tests.test_histogram.TestHistogram testMethod=test_bincount_invalid_weight_length>
x_type = <class 'numpy.float16'>, w_type = <class 'numpy.uint16'>

    @for_all_dtypes_combination_bincount(names=['x_type', 'w_type'])
    def test_bincount_invalid_weight_length(self, x_type, w_type):
        for xp in (numpy, cupy):
            x = testing.shaped_arange((1,), xp, x_type)
            w = testing.shaped_arange((2,), xp, w_type)
            with pytest.raises(ValueError):
>               xp.bincount(x, weights=w)

tests/cupy_tests/statics_tests/test_histogram.py:274: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (array([1.], dtype=float16),)
kwargs = {'weights': array([1, 2], dtype=uint16)}
relevant_args = (array([1.], dtype=float16), array([1, 2], dtype=uint16))

>   ???
E   TypeError: Cannot cast array data from dtype('float16') to dtype('int64') according to the rule 'safe'

<__array_function__ internals>:6: TypeError
----------------------------- Captured stdout call -----------------------------
{'x_type': <class 'numpy.float16'>, 'w_type': <class 'numpy.uint16'>}
```

**Comment by asi1024** [comment]
Duplicates to #3256.

**Comment by rushabh-v** [comment]
Oh, I am closing this.

**Pull request closed without merging**